cinema/views.py

```diff
@@ -1 +1,62 @@
-# write views here
+from rest_framework import viewsets
+from cinema.models import CinemaHall, Genre, Actor, Movie, MovieSession
+from cinema.serializers import (
+    CinemaHallSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    MovieSerializer,
+    MovieListSerializer,
+    MovieRetrieveSerializer,
+    MovieSessionSerializer,
+    MovieSessionRetrieveSerializer,
+    MovieSessionListSerializer,
+)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all().prefetch_related("genres", "actors")
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+    serializer_class = MovieSessionSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionListSerializer
+        elif self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+
+        return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ["list", "retrieve"]:
+            queryset = queryset.prefetch_related("movie", "cinema_hall")
```

Comment:
In the `get_queryset` method, consider using `select_related` instead of `prefetch_related` for `movie` and `cinema_hall`, as they are `ForeignKey` fields. This will improve performance by reducing the number of queries.